### PR TITLE
Make referral banner dismissible

### PR DESCRIPTION
**Pull request by gijoe0295**
### Details

This PR makes referral banner dismissible **temporarily**.

### Fixed Issues

$ https://github.com/Expensify/App/issues/32499
PROPOSAL: https://github.com/Expensify/App/issues/32499#issuecomment-1841355500

### Tests

1. Press FAB >> Start chat
2. Press close referral banner
3. Verify the banner is closed
4. Verify the text input remains focused
5. Close the RHP
6. Repeat step 1
7. Verify the banner appears again
8. Repeat those steps with Request money and Search page

- [x] Verify that no errors appear in the JS console

### Offline tests

NA

### QA Steps

1. Press FAB >> Start chat
2. Press close referral banner
3. Verify the banner is closed
4. Verify the text input remains focused
5. Close the RHP
6. Repeat step 1
7. Verify the banner appears again
8. Repeat those steps with Request money and Search page

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/153004152/2b1958a6-968a-45b1-b201-224669d4e5ac

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/153004152/478bef0d-0e4b-4748-8d2c-3c7f3c4c51d6

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/153004152/3def7fba-a737-462b-9be4-23b8e82b17b1

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/153004152/33c8aeeb-2bdc-497d-a277-e4a97455df88

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/153004152/39883e97-bd95-47b5-9d02-e2383bf532f8

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/153004152/e8ab0770-16cb-41b8-91e4-6a318b084108

</details>

### CODE REVIEW

**Review by dannymcclain — DISMISSED** [review]
Looks to be working as expected to me. From a design perspective I think we're all good, but please make sure to still get a review from someone with a more technical background. 👍

**Comment by abdulrahuman5196** [comment]
Hi, Will review today.

**Comment by abdulrahuman5196** [comment]
Reviewing now

File: src/components/OptionsSelector/BaseOptionsSelector.js
Comment (abdulrahuman5196) [inline src/components/OptionsSelector/BaseOptionsSelector.js:649]: It seems the pressable and the icon have same size.


https://github.com/Expensify/App/assets/46707890/f7abc94f-16df-4118-bb6b-1f40a5f253a3

Shouldn't we have some extra size for pressing? For now both the icon size and the pressing place size is 20

@dannymcclain yours thoughts on this?

Comment (dannymcclain) [reply src/components/OptionsSelector/BaseOptionsSelector.js:649]: Good call—ideally the pressable area of the `x` icon should be at least a 44px square.

Comment (shawnborton) [reply src/components/OptionsSelector/BaseOptionsSelector.js:649]: I agree with that Danny. But I could have sworn we can also solve for this using React Native's [hitslop](https://reactnative.dev/docs/touchablewithoutfeedback#hitslop), right? This we we can keep the button where it is but just define a greater hit area around around the button.

**Comment by tgolen** [comment]
What's the next step on this PR? I am beginning to work on https://github.com/Expensify/App/issues/34387 which is blocked on the functionality in this PR. I'll start working off the branch, but I also want to encourage everyone to work with urgency to get this completed. It should be reviewed or updated every day and it's been 4 days with no activity.

**Comment by abdulrahuman5196** [comment]
Hi, it's miss from my end. Will test and update first thing in my morning and close this out ASAP.

**Comment by abdulrahuman5196** [comment]
Reviewing now

File: src/pages/SearchPage/SearchPageFooter.tsx
Comment (tgolen) [inline src/pages/SearchPage/SearchPageFooter.tsx:13]: I don't think this works, does it? I think there is some confusion with the `SearchPage` component: https://expensify.slack.com/archives/C01GTK53T8Q/p1705959664098999?thread_ts=1701232130.505309&cid=C01GTK53T8Q

Comment (tgolen) [reply src/pages/SearchPage/SearchPageFooter.tsx:13]: OK, looks like the old file `SearchPage.js` will be getting removed. Thanks!

**Comment by tgolen** [comment]
Bump @abdulrahuman5196 for review, please. Let's get this one finished up.

**Comment by abdulrahuman5196** [comment]
Working on another critical PR yesterday and today. Will close out that and start in 30 minutes. Sorry for the delay

**Comment by abdulrahuman5196** [comment]
Starting to review now

**Comment by abdulrahuman5196** [comment]
NAB: @gijoe0295 Kindly update the Tests in author's checklist to reflect all the pages which has this change.

**Comment by abdulrahuman5196** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>




https://github.com/Expensify/App/assets/46707890/ae8c9471-f152-424b-9f2e-4039c13389db



</details>

<details>
<summary>Android: mWeb Chrome</summary>




https://github.com/Expensify/App/assets/46707890/17caf0e2-9380-4ec4-ad91-9a7d8f023ac2



</details>

<details>
<summary>iOS: Native</summary>






https://github.com/Expensify/App/assets/46707890/bdc913d0-42fa-42af-8533-8c1a0fd50c69






</details>

<details>
<summary>iOS: mWeb Safari</summary>




https://github.com/Expensify/App/assets/46707890/10317e81-22ba-405d-b35e-bfd0046ab936



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>




https://github.com/Expensify/App/assets/46707890/f04a7444-0558-4a09-a1ed-7aa4b473999b


</details>

<details>
<summary>MacOS: Desktop</summary>




https://github.com/Expensify/App/assets/46707890/15800364-dfc9-4fe3-84bf-a16970174189



</details>

**Review by abdulrahuman5196 — APPROVED** [review]
Changes looks good and works well. Reviewers checklist is also complete.

All yours. @tgolen / @joelbettner 

🎀 👀 🎀
C+ Reviewed

**Comment by abdulrahuman5196** [comment]
@gijoe0295 Kindly also update the testing screenshots after high number of changes, since we added changes in couple more pages after last update.

**Review by tgolen — APPROVED** [review]

**Comment by tgolen** [comment]
All yours @joelbettner

**Review by joelbettner — APPROVED** [review]

**Pull request merged**
